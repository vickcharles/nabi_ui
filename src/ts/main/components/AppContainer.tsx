import * as React from 'react';
import { Dispatch } from 'redux';
import { connect } from 'react-redux';
import { Route, withRouter, Switch } from 'react-router-dom';

import Header from './Header';
import Homepage from './Homepage';
import { createUser, changeAvatar, Registration } from '../../Users/';
import { UserState } from '../../Users/model';
import { createInstructor, ProfileBuilder } from '../../Instructors';
import { InstructorState } from '../../Instructors/model';

export interface AppContainerStateProps {
  users: UserState[];
  dispatch: Dispatch<{}>;
}

export interface AppContainerProps extends AppContainerStateProps {}

class AppContainer extends React.Component<AppContainerProps, {}> {
  public render(): JSX.Element {
    
    const { dispatch, users } = this.props;

    const dispatchCreateUser: any = (user: UserState) => dispatch(createUser(user));
    const dispatchCreateInstructor: any = (instructor: InstructorState) => dispatch(createInstructor(instructor));
    const dispatchChangeAvatar: any = (id: string, avatar: string) => dispatch(changeAvatar(id, avatar ));
    
    const renderRegistration = (props: any) => (
      <Registration 
        createUser={dispatchCreateUser} 
        createInstructor={dispatchCreateInstructor}
      />
    );

    const renderProfileBuilder = (props: any) => (
      <ProfileBuilder 
        users={users} 
        classes={null}
        changeAvatar={dispatchChangeAvatar}
      />
    );

    return (
      <div>
        <Header />
        <Switch>
          <Route 
            exact={true} 
            path="/" 
            component={Homepage} 
          />
          <Route 
            exact={true} 
            path="/registration" 
            render={renderRegistration}
          />
          <Route 
            exact={true} 
            path="/profile-builder/:id" 
            render={renderProfileBuilder}
          />
        </Switch>
      </div>
    );
  }
}

const mapStateToProps = (state: any, _ownProps: any) => ({
  users: state.users
});

export default withRouter(connect(mapStateToProps)(AppContainer));
