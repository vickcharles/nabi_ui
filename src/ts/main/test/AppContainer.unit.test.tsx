import * as React from 'react';
import AppContainer from '../components/AppContainer';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';

describe('AppContainer', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(
      <MemoryRouter>
        <AppContainer />
      </MemoryRouter>
    ).find(AppContainer).dive();
  });

  it('Renders the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
