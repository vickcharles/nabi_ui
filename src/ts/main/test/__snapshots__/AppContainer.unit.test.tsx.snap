// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer Renders the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Connect(AppContainer)) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
